doc/craft/ratelimitpolicy.md

```diff
@@ -0,0 +1,255 @@
+`matches` defines the predicate used to match requests to a given action.
+Multiple match types are ANDed together, i.e. the match will evaluate to true only if all conditions are satisfied.
+A match can be made up of 4 optional fields, `headers`, `method`, `path` and `queryParams`
+
+- `headers`: a list of objects referencing HTTP request headers.
+Formatted as follows: `{name: <header name>, value: <expected value> }`.
+- `method`: targets the HTTP method of the request: i.e. GET, HEAD, POST, PUT, DELETE, CONNECT, OPTIONS, TRACE, PATCH
+- `path`: an object that denotes the type of match and the path. 
+Formatted as follows: `{type: <match type>, value: <URI>}`.
+Examples of the `type` are: Exact, PathPrefix, RegularExpression.
+The default behaviour if the `path` is not specified if `{type: PathPrefix, value: /}`.
+- `queryParams`: describes how to select a HTTP route by matching HTTP query parameters.
+The object is formatted like `{name: <param name>, value: <param value>, type: <Exact|RegularExpression>}`.
+    - `name`: minimum length = 1, maximum length = 256
+    - `value`: minimum length = 1, maximum length = 1024
+    - `type`: optional, accepted values: Exact, RegularExpression.
+    Default value: Exact
+
+```yaml
+spec:
+    limits:
+        limit_1:
+            routeSelectors:
+                  matches:
+                    - method: GET
+                      path:
+                        type: PathPrefix
+                        value: /admin
+                      headers:
+                        - name: "version"
+                          value: "v1"
+                      queryParam:
+                        - name: "search"
+                          value: "release"
+```
+The above example selects all HTTP requests that have a method of GET, and the URI starts with "/admin".
+Also the request must have a header entry of `version:v1` and a queryParam of `search=release`.
+All fields are optional, and the above example shows how complex of a match that can be crafted.
+
+#### when
+`when` defines semantics for matching an HTTP request based on conditions.
+This should be used only when no match for the request can be created using [routeSelectors](#routeSelectors).
+Three fields used are: operator, selector, value
+
+- `operator`: binary operator to be applied to the content fetched from the selector.
+Possible values are: "eq" (equal to), "neq" (not equal to).
+- `selector`: defines one item from the well known attributes.
+The full list of well known attributes can be found [here](/architecture/rfcs/0002-well-known-attributes/#request-attributes).
+- `value`: reference for the comparison.
+```yaml
+spec:
+    limits:
+        limit_1:
+            when:
+                - operator: "eq"
+                  selector: "request.useragent"
+                  value: "curl/8.0.1"
+```
+The above example will match HTTP requests that are sent with the header HTTP_USER_AGENT set with a value of "curl/8.0.1". 
+
+### targetRef
```

Comment:
Maybe explain this before explaining `routeSelectors`?

Reply:
I explained the different sections in an alphabetically order as this is how a user would see items listed in a CR.  If this was to be ordered by the perceived importances should `routeSelectors` be listed before `counters`?